default/python/AI/ShipDesignAI.py

```diff
@@ -208,7 +206,7 @@ def print_parts_for_planets(self, pid=None):
         elif isinstance(pid, list):
             planets = pid
         else:
-            print >> sys.stderr, "Invalid parameter 'pid' for 'print_parts_for_planets'. Expected int, list or None."
+            warn("Invalid parameter 'pid' for 'print_parts_for_planets'. Expected int, list or None.")
```

Comment:
This should be an error.  The type of `pid `is wrong, which should never happen.